x/concentrated-liquidity/incentives.go

```diff
@@ -592,8 +645,20 @@ func (k Keeper) createIncentive(ctx sdk.Context, poolId uint64, sender sdk.AccAd
 		MinUptime:        minUptime,
 	}
 
+	// Get all incentive records for uptime
+	existingRecordsForUptime, err := k.GetAllIncentiveRecordsForUptime(ctx, poolId, minUptime)
+	if err != nil {
+		return types.IncentiveRecord{}, err
+	}
+
+	// Fixed gas consumption per swap to prevent spam
+	ctx.GasMeter().ConsumeGas(uint64(types.BaseGasFeeForNewIncentive*len(existingRecordsForUptime)), "balancer swap computation")
```

Comment:
Resolved offline. Maybe we can add this comment here; 
```
we charge for # of uptime records of all kinds, so it implicitly covers DoS subvectors for any specific field
e.g. different start times, different emission rates, different remaining amounts etc.
All implicitly covered since the bottleneck is the specific uptime we’re scaling the gas cost on
```